### PR TITLE
FIX: do not try to render empty images

### LINKED ISSUE #7886: empty imshow crashes python
To help us understand and resolve your issue, please fill out the form to the best of your ability. You can feel free to delete the sections that do not apply.

### Bug report

**Bug summary**

- A short 1-2 sentences that succinctly describes the bug

An empty array in imshow + an extent crashes 

**Code for reproduction**

- A minimum code snippet required to reproduce the bug, also minimizing the number of dependencies required

```python
import matplotlib.pyplot as plt
plt.ioff()
fig, ax = plt.subplots()
im = ax.imshow([[]])
im.set_extent([-5, 5, -5, 5])
fig.canvas.draw()
```

**Actual outcome**

- The output produced by the above code, which may be a screenshot, console output, etc.

```
22:51 $ python /tmp/st.py 
/home/tcaswell/src/p/matplotlib/lib/matplotlib/axes/_base.py:2903: UserWarning: Attempting to set identical left==right results
in singular transformations; automatically expanding.
left=-0.5, right=-0.5
  'left=%s, right=%s') % (left, right))
/home/tcaswell/src/p/matplotlib/lib/matplotlib/image.py:327: RuntimeWarning: divide by zero encountered in double_scalars
  in_bbox.width / A.shape[1],
Floating point exception (core dumped)

```

**Expected outcome**

- A description of the expected outcome from the code snippet
 not crash
- If this used to work in an earlier version of Matplotlib, please note the version it used to work on
  1.5.3 (reportedly, not verified)

- Matplotlib version, Python version and Platform (Windows, OSX, Linux ...)
 py3.6, pyqt4, linux.  

- How did you install Matplotlib and Python (pip, anaconda, from source ...)
 source :smiling_imp:

### PR DESCRIPTION
**Pull request by tacaswell**
Do not bother trying to render Images which have zero-dimension in any
direction.  Simply return from the `draw` method early

closes #7886

### CODE REVIEW

**Comment by tacaswell** [comment]
@NelleV Did end up adding an exception, but should almost always be hidden behind a short-circuit.

File: lib/matplotlib/image.py [hunk 1]
```diff
@@ -478,9 +482,17 @@ def _check_unsampled_image(self, renderer):
 
     @allow_rasterization
     def draw(self, renderer, *args, **kwargs):
+        # if not visible, declare victory and return
         if not self.get_visible():
+            self.stale = False
             return
 
+        # for empty images, there is nothing to draw!
+        if any(s == 0 for s in self.get_size()):
```
Comment (anntzer) [inline lib/matplotlib/image.py]: self.get_array().size == 0

**Comment by NelleV** [comment]
It looks good to me. It's a better fix than the one I had :)

File: lib/matplotlib/image.py [hunk 2]
```diff
@@ -488,7 +488,7 @@ def draw(self, renderer, *args, **kwargs):
             return
 
         # for empty images, there is nothing to draw!
-        if any(s == 0 for s in self.get_size()):
+        if self.get_arrayy().size == 0:
```
Comment (QuLogic) [inline lib/matplotlib/image.py]: Extra 'y'.

**Pull request merged**